### PR TITLE
ref(dropdown-menu-v2): Use Link component for menu links

### PR DESCRIPTION
**Pull request by vuluongj20**
Right now, inside `DropdownMenuV2`, we procedurally handle links with the router object (via `router.push(item.to)`). This simulates the behavior of anchor elements, but not fully. If the user command-clicks on the menu item, it will open the link in the same tab (not in a new tab as expected).

Replacing the procedural call with an actual `<Link />` component fixes this. This, however, doesn't work when the user navigates using the keyboard (i.e. when they press `Enter`). To make it work, we can add a keyboard event listener to listen for the `Enter` key, and trigger the link manually.

Screen Shot 2022-04-07 at 5 34 45 PM

### CODE REVIEW

File: static/app/components/dropdownMenuItemV2.tsx [hunk 1]
```diff
@@ -126,171 +125,162 @@ type Props = {
...
+    onKeyDown: e => {
+      if (e.key === 'Enter' && item.to) {
+        const mouseEvent = new MouseEvent('click', {
+          ctrlKey: e.ctrlKey,
+          metaKey: e.metaKey,
+        });
+        ref.current?.querySelector(`${InnerWrap}`)?.dispatchEvent(mouseEvent);
+        onClose();
+        return;
       }
```
Comment (vuluongj20) [inline static/app/components/dropdownMenuItemV2.tsx:186]: if this item is a link, and if the user pressed `Enter`, then manually trigger the link

File: static/app/components/dropdownMenuItemV2.tsx [hunk 2]
```diff
@@ -126,171 +125,162 @@ type Props = {
...
-          return;
-        }
-        state.selectionManager.clearSelection();
+  // Open submenu on arrow right key press
+  const {keyboardProps} = useKeyboard({
+    onKeyDown: e => {
+      if (e.key === 'Enter' && item.to) {
+        const mouseEvent = new MouseEvent('click', {
+          ctrlKey: e.ctrlKey,
+          metaKey: e.metaKey,
```
Comment (vuluongj20) [inline static/app/components/dropdownMenuItemV2.tsx:181]: carry modifier keys (`command`/`ctrl`) over to the synthetic event. This means that if the user presses `command`/`ctrl` + `Enter`, then the link will open in a new tab.

File: static/app/components/dropdownMenuItemV2.tsx [hunk 3]
```diff
@@ -315,13 +305,33 @@ const getHoverBackground = (theme: Theme, priority?: Priority) => {
...
+  ${p => p.priority === 'danger' && `&,&:hover {color: ${p.theme.errorText}}`}
+  ${p =>
+    p.isDisabled &&
+    `
+    &, &:hover {
+      color: ${p.theme.subText};
+      cursor: initial;
+    }
+  `}
+
```
Comment (vuluongj20) [inline static/app/components/dropdownMenuItemV2.tsx:334]: needed to override the default blue link color

**Review by edwardgou-sentry — APPROVED** [review]
Code looks good to me, tested and it works correctly.
Created a pr based off this branch here to also update the Discover `Add to Dashboard` links to properly utilize the `to` prop when the user is flagged with the new widget builder feature flag:
https://github.com/getsentry/sentry/pull/33506

**Pull request merged**